sparkmagic/sparkmagic/kernels/kernelmagics.py

```diff
@@ -331,12 +331,15 @@ def _do_not_call_change_language(self, line, cell="", local_ns=None):
 
     @magic_arguments()
     @line_magic
+    @argument("--authentication", type=str, default=NONE_AUTH, choices=POSSIBLE_AUTH, \
```

Comment:
Please also update the server extension to pass in the authentication method:
https://github.com/jupyter-incubator/sparkmagic/blob/67fcf744b94f5c96eea26a8729a0f004576cdcf0/sparkmagic/sparkmagic/serverextension/handlers.py#L55

Look at `_get_kernel_name` for how the server extension handles optional parameters. I think default should be configurable, with default being `BASIC_AUTH`.